test/options.jl

```diff
@@ -284,6 +284,63 @@
         @test fmt(str_, 4, length(str_) - 1, short_to_long_function_def = true) == str
     end
 
+    @testset "function longdef to shortdef" begin
+        str_ = """
+        function foo(a)
+            bodybodybody
+        end"""
+        str = "foo(a) = bodybodybody"
+        @test fmt(str_, 4, length(str) - 1, long_to_short_function_def = true) == str_
+        @test fmt(str_, 4, length(str), long_to_short_function_def = true) == str
+
+        str_ = """
+        function foo(a::T) where {T}
+            bodybodybodybodybodybodyb
+        end"""
+        str = "foo(a::T) where {T} = bodybodybodybodybodybodyb"
+        @test fmt(str_, 4, length(str) - 1, long_to_short_function_def = true) == str_
+        @test fmt(str_, 4, length(str), long_to_short_function_def = true) == str
+
+        str_ = """
+        function foo(a::T)::R where {T}
+            bodybodybodybodybodybodybody
+        end"""
+        str = "foo(a::T)::R where {T} = bodybodybodybodybodybodybody"
+        @test fmt(str_, 4, length(str) - 1, long_to_short_function_def = true) == str_
+        @test fmt(str_, 4, length(str), long_to_short_function_def = true) == str
+
+        str_ = """
+        function foo(a)
+            return a + 1
+        end"""
+        str = "foo(a) = a + 1"
+        @test fmt(str_, 4, length(str), long_to_short_function_def = true) == str
+
+        str = """
+        function foo()
+            expr1
+            expr2
+        end"""
+        @test fmt(str, 4, length(str), long_to_short_function_def = true) == str
+
+        str_ = """
+        function foo(a)
+            return if a > 1
+                2
+            else
+                nothing
+            end
+        end"""
+        str = """
+        foo(a) =
+            if a > 1
+                2
+            else
+                nothing
+            end"""
+        @test fmt(str_, 4, length(str), long_to_short_function_def = true) == str
```

Comment:
same here, is it just that if the margin is big it will go to the short def kind of thing?

Reply:
So this one is a bit more tricky. I wished it could be transformed into:
```julia
foo(a)  = if a > 1
    2
else
    nothing
end
```
in order to same another extra line, but I couldn't find a fix regarding the addition of newlines in https://github.com/domluna/JuliaFormatter.jl/blob/598a2485bbd36a118c66b3774e73684c4a4e9697/src/fst.jl#L647.

However, the current (saving one line over the long def) is imo satisfactory enough.

Reply:
yeah I guess it would be the inverse of `conditional_to_if`

Reply:
Maybe I'll have some time to implement `if_to_conditional` later. I'll open an issue if someone wants to help.

Reply:
hmm  could you just replace that Newline node with Whitespace(0) ?

Reply:
either way this is not a big deal, it's a subjective addition

Reply:
> hmm could you just replace that Newline node with Whitespace(0) 

I tried that, but it didn't work :/